src/routes/write-exercise/components/WriteExercise.tsx

```diff
@@ -41,14 +42,23 @@ const WriteExercise = ({ documents, route, navigation }: WriteExercisePropType):
     }))
   )
 
+  const keyboard = useKeyboard()
   const current = documentsWithResults[currentIndex]
   const nthRetry = current.numberOfTries ?? 0
   const needsToBeRepeated = nthRetry < numberOfMaxRetries && current.result !== SIMPLE_RESULTS.correct
 
   const tryLater = useCallback(() => {
-    setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
-    Keyboard.dismiss()
-  }, [documentsWithResults, currentIndex])
+    // Wait until keyboard is hidden before go next
+    if (keyboard.keyboardShown) {
+      const onKeyboardHideSubscription = Keyboard.addListener('keyboardDidHide', () => {
+        setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
+        onKeyboardHideSubscription.remove()
+      })
+      Keyboard.dismiss()
+    } else {
+      setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
+    }
```

Comment:
Why not just:
```suggestion
    if (keyboard.keyboardShown) {
      Keyboard.dismiss()
    }
    setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
```

Reply:
because the dismiss is not fast enough and the next screen gets rerendered with minimized picture and then the keyboard closes